test/suites/scale/deprovisioning_test.go

```diff
@@ -0,0 +1,631 @@
+				}
+				deploymentOptions.PodOptions.Tolerations = []v1.Toleration{
+					{
+						Key:      deprovisioningTypeKey,
+						Operator: v1.TolerationOpEqual,
+						Value:    v,
+						Effect:   v1.TaintEffectNoSchedule,
+					},
+				}
+				deploymentOptions.Replicas = int32(replicas)
+				d := test.Deployment(deploymentOptions)
+				deploymentMap[v] = d
+			}
+
+			provisionerMap := map[string]*v1alpha5.Provisioner{}
+			// Generate all the provisioners for multi-deprovisioning
+			for _, v := range deprovisioningTypes {
+				provisionerOptions.Taints = []v1.Taint{
+					{
+						Key:    deprovisioningTypeKey,
+						Value:  v,
+						Effect: v1.TaintEffectNoSchedule,
+					},
+				}
+				provisionerOptions.Labels = map[string]string{
+					deprovisioningTypeKey: v,
+				}
+				provisionerOptions.Kubelet = &v1alpha5.KubeletConfiguration{
+					MaxPods: lo.ToPtr[int32](int32(maxPodDensity)),
+				}
+				provisionerMap[v] = test.Provisioner(provisionerOptions)
+			}
+
+			By("waiting for the deployment to deploy all of its pods")
+			var wg sync.WaitGroup
+			for _, d := range deploymentMap {
+				wg.Add(1)
+				go func(dep *appsv1.Deployment) {
+					defer GinkgoRecover()
+					defer wg.Done()
+
+					env.ExpectCreated(dep)
+					env.EventuallyExpectPendingPodCount(labels.SelectorFromSet(dep.Spec.Selector.MatchLabels), int(lo.FromPtr(dep.Spec.Replicas)))
+				}(d)
+			}
+			wg.Wait()
+
+			// Create a separate nodeTemplate for drift so that we can change the nodeTemplate later without it affecting
+			// the other provisioners
+			driftNodeTemplate := nodeTemplate.DeepCopy()
+			driftNodeTemplate.Name = test.RandomName()
+			provisionerMap[driftValue].Spec.ProviderRef.Name = driftNodeTemplate.Name
+
+			env.MeasureDurationFor(func() {
+				By("kicking off provisioning by applying the provisioner and nodeTemplate")
+				env.ExpectCreated(driftNodeTemplate, nodeTemplate)
+				for _, p := range provisionerMap {
+					env.ExpectCreated(p)
+				}
+
+				env.EventuallyExpectCreatedMachineCount("==", expectedNodeCount)
+				env.EventuallyExpectCreatedNodeCount("==", expectedNodeCount)
+				env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+
+				// Wait for all pods across all deployments we have created to be in a healthy state
+				wg = sync.WaitGroup{}
+				for _, d := range deploymentMap {
+					wg.Add(1)
+					go func(dep *appsv1.Deployment) {
+						defer GinkgoRecover()
+						defer wg.Done()
+
+						env.EventuallyExpectHealthyPodCount(labels.SelectorFromSet(dep.Spec.Selector.MatchLabels), int(lo.FromPtr(dep.Spec.Replicas)))
+					}(d)
+				}
+				wg.Wait()
+			}, aws.ProvisioningEventType, multipleDeprovisionersTestGroup, defaultTestName)
+
+			env.Monitor.Reset() // Reset the monitor so that we now track the nodes starting at this point in time
+
+			By("scaling down replicas across deployments")
+			deploymentMap[consolidationValue].Spec.Replicas = lo.ToPtr[int32](int32(int(float64(replicas) * 0.2)))
```

Comment:
Thoughts on just scaling this down by 50%? Should be easier to assert a number of nodes after consolidation with a factor of 2 rather than 5, considering that instance type sizes also change by a factor of 2 as well.

Reply:
I'm not convinced that it really matters one way or the other. I think we will eventually do different levels of utilization anyways so starting with 20% seems reasonable to me.